plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -652,6 +654,25 @@ public void testOptimizingPartitionsOfV2TableWithGlobalEqualityDeleteFile()
                         .toArray(String[]::new));
     }
 
+    @Test
+    public void testOptimizingWholeTableRemovesDeleteFiles()
+            throws Exception
+    {
+        try (TestTable testTable = newTrinoTable("test_optimize_removes_obsolete_delete_files_", "AS SELECT * FROM tpch.tiny.nation")) {
+            assertUpdate("DELETE FROM " + testTable.getName() + " WHERE regionkey % 2 = 0", 15);
+            Table icebergTable = updateTableToV2(testTable.getName());
+            writeEqualityDeleteToNationTable(icebergTable, Optional.of(icebergTable.spec()), Optional.of(new PartitionData(new Long[] {1L})));
+
+            assertQuery("SELECT count(*) FROM \"" + testTable.getName() + "$files\" WHERE content = " + POSITION_DELETES.id(), "VALUES 1");
+            assertQuery("SELECT count(*) FROM \"" + testTable.getName() + "$files\" WHERE content = " + EQUALITY_DELETES.id(), "VALUES 1");
+
+            assertQuerySucceeds("ALTER TABLE " + testTable.getName() + " EXECUTE OPTIMIZE");
+
+            assertQuery("SELECT count(*) FROM \"" + testTable.getName() + "$files\" WHERE content = " + POSITION_DELETES.id(), "VALUES 0");
+            assertQuery("SELECT count(*) FROM \"" + testTable.getName() + "$files\" WHERE content = " + EQUALITY_DELETES.id(), "VALUES 0");
```

Comment:
Can we verify the table result (`"SELECT * FROM " + testTable.getName()`) as well?